From 9d4398032a1c5618819c82b59ed6d18151072937 Mon Sep 17 00:00:00 2001
From: Daniel Ammann <daniel.ammann@bytesatwork.ch>
Date: Tue, 7 Mar 2023 08:08:47 +0100
Subject: [PATCH] board: bytedevkit: Allow environment on mmc even when booting
 from nor

Minimize nor flash wear by not writing to it unnecessarily.

Signed-off-by: Daniel Ammann <daniel.ammann@bytesatwork.ch>
---
 board/bytesatwork/bytedevkit/stm32mp1.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/board/bytesatwork/bytedevkit/stm32mp1.c b/board/bytesatwork/bytedevkit/stm32mp1.c
index 5dff8faecce..7813e0dd16c 100644
--- a/board/bytesatwork/bytedevkit/stm32mp1.c
+++ b/board/bytesatwork/bytedevkit/stm32mp1.c
@@ -448,6 +448,7 @@ enum env_location env_get_location(enum env_operation op, int prio)
 
 	switch (bootmode & TAMP_BOOT_DEVICE_MASK) {
 #ifdef CONFIG_ENV_IS_IN_MMC
+	case BOOT_FLASH_NOR:
 	case BOOT_FLASH_SD:
 	case BOOT_FLASH_EMMC:
 		return ENVL_MMC;
@@ -499,7 +500,7 @@ int mmc_get_env_dev(void)
 {
 	u32 bootmode = get_bootmode();
 
-	return (bootmode & TAMP_BOOT_INSTANCE_MASK) - 1;
+	return CONFIG_SYS_MMC_ENV_DEV;
 }
 #endif
 
-- 
2.39.2

